flash_attn/cute/flash_bwd_sm100.py

```diff
@@ -320,11 +320,11 @@ def _setup_smem_layout(self):
         )
         self.sdKV_epi_tile = (
             self.tile_n,
-            128 // (self.dk_dtype.width // 8),  # 64 or 32
+            min(128 // (self.dk_dtype.width // 8), self.tile_hdim // 2),  # 64 or 32
         )  # subtiles mma_tiler_dsq[:2] = mma_tiler_pdo[:2]
-        self.num_epi_stages = (self.tile_hdim // 2) // self.sdKV_epi_tile[1]
+        # headdim_64 gets 1 stage
+        self.num_epi_stages = max(1, (self.tile_hdim // 2) // self.sdKV_epi_tile[1])
```

Comment:
if this was still 2 stages at headdim64 for gqa, we would not run into buffer issues i.e. `epi_tile = ... self.tile_hdim // (self.dk_dtype.width // 8)`, right?

Reply:
uhmm hmm so I think because we set `self.dK_reduce_ncol = 32` this wont naively work, and instead it would need to be 16. and then we might be able to split the tile and do each sub write. 

Which buffer issue though are you referring to though? there has been a few 😂